core/capabilities/ccip/ccipaptos/extradatadecoder.go

```diff
@@ -0,0 +1,201 @@
+package ccipaptos
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+	ccipcommon "github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/common"
+)
+
+// ExtraDataDecoder is a concrete implementation of ccipcommon.ExtraDataDecoder
+// Compatible with ccip::fee_quoter version 1.6.0
+type ExtraDataDecoder struct{}
+
+var _ ccipcommon.ExtraDataDecoder = (*ExtraDataDecoder)(nil)
+
+const (
+	aptosDestExecDataKey = "destGasAmount"
+)
+
+var (
+	// bytes4 public constant EVM_EXTRA_ARGS_V1_TAG = 0x97a657c9;
+	evmExtraArgsV1Tag = hexutil.MustDecode("0x97a657c9")
+
+	// bytes4 public constant GENERIC_EXTRA_ARGS_V2_TAG = 0x181dcf10;
+	genericExtraArgsV2Tag = hexutil.MustDecode("0x181dcf10")
+
+	// bytes4 public constant SVM_EXTRA_EXTRA_ARGS_V1_TAG = 0x1f3b3aba
+	svmExtraArgsV1Tag = hexutil.MustDecode("0x1f3b3aba")
+
+	uint32Type     = mustNewType("uint32")
+	uint64Type     = mustNewType("uint64")
+	uint256Type    = mustNewType("uint256")
+	boolType       = mustNewType("bool")
+	bytes32Type    = mustNewType("bytes32")
+	bytes32ArrType = mustNewType("bytes32[]")
+
+	// Arguments for decoding destGasAmount
+	destGasAmountArguments = abi.Arguments{
+		{Name: aptosDestExecDataKey, Type: uint32Type},
+	}
+
+	// Arguments matching the fields of EVMExtraArgsV1 struct
+	evmExtraArgsV1Fields = abi.Arguments{
+		{Name: "gasLimit", Type: uint256Type},
+	}
+
+	// Arguments matching the fields of GenericExtraArgsV2 struct
+	genericExtraArgsV2Fields = abi.Arguments{
+		{Name: "gasLimit", Type: uint256Type},
+		{Name: "allowOutOfOrderExecution", Type: boolType},
+	}
+
+	// Arguments matching the fields of SVMExtraArgsV1 struct
+	svmExtraArgsV1Fields = abi.Arguments{
+		{Name: "computeUnits", Type: uint32Type},
+		{Name: "accountIsWritableBitmap", Type: uint64Type},
+		{Name: "allowOutOfOrderExecution", Type: boolType},
+		{Name: "tokenReceiver", Type: bytes32Type},
+		{Name: "accounts", Type: bytes32ArrType},
+	}
+)
+
+func mustNewType(typeStr string) abi.Type {
+	t, err := abi.NewType(typeStr, "", nil)
+	if err != nil {
+		panic(fmt.Sprintf("failed to create ABI type %s: %v", typeStr, err))
+	}
+	return t
+}
+
+// DecodeDestExecDataToMap reformats bytes into a chain agnostic map[string]interface{} representation for dest exec data
+func (d ExtraDataDecoder) DecodeDestExecDataToMap(destExecData cciptypes.Bytes) (map[string]any, error) {
+	args := make(map[string]any)
+	err := destGasAmountArguments.UnpackIntoMap(args, destExecData)
```

Comment:
Kinda confused about this, we're using ethereum abi-encoding on Aptos? From what I can see this method is used in `DecodeTokenAmountDestExecData` which gets provided a `sourceChainSelector`. So for example, if we send a msg from Aptos -> EVM, Aptos is the src chain in this case, and we would try to decode the `destExecData` emitted from the Aptos CCIPMessageSent event.